### ISSUE TITLE
10-7959f-1 - Pre-fill not working for logged in users

### ISSUE THREAD

**Issue by marywang2** [department-of-veterans-affairs/va.gov-team#81646]
**Issue**: For the logged in user viewing 10-7959f-1 on staging, the information from their profile is not pre-filled in the form fields.
**Current result**: the fields are blank and the user has to manually fill in
**Expected result:** 
- the fields would be pre-filled with their profile information when they are signed in 
- a logged in user should also be able to save their progress if they leave the form without completing.

**Comment by marywang2** [comment]
Related to this [back end ticket](https://app.zenhub.com/workspaces/ivc-forms-652da2d3f0ae4c0016bfb109/issues/gh/department-of-veterans-affairs/va.gov-team/80367)

**Comment by balexandr** [comment]
@reiting Here is the mapping I created in the backend:
```
veteran:
  dateOfBirth: [identity_information, date_of_birth]
  fullName: [identity_information, full_name]
  first: [identity_information, first]
  middle: [identity_information, middle]
  last: [identity_information, last]
physicalAddress:
  street: [contact_information, address, street]
  street2: [contact_information, address, street2]
  city: [contact_information, address, city]
  state: [contact_information, address, state]
  country: [contact_information, address, country]
  postalCode: [contact_information, address, postal_code]
veteranSocialSecurityNumber: [identity_information, ssn]
phoneNumber: [contact_information, us_phone]
emailAddress: [contact_information, email]
```

I know we have mailingAddress but the backend isn't set up at all to handle two sets of addresses so let's just get this working and we can work on that later if need be

**Comment by reiting** [comment]
@marywang2 this is waiting for product review right now. I don't think we can really test this locally, as apparently it doesn't really work on local. Not super ideal, but it does match up with Bryan's backend as far as we can tell, and hopefully this along with what he's done together can come together on staging!

**Comment by marywang2** [comment]
@balexandr @reiting I'm not seeing the changes on staging, let me know if we should look together.

**Comment by michaelclement** [comment]
I should have caught this in the PR review earlier, but it looks like we're accessing the keys from the original backend configuration instead of the current configuration. For example, currently on the backend we have this:
<img width="589" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWk2QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--76ade6e6edf440d8ebd3d0da7256aec24180511f/backend.png" alt="backend.png" />

But then on the front end in the prefill transformer we access this:
<img width="724" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMW02QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d33a27ca1df718451cf8ba38d638405dea80cf25/frontend.png" alt="frontend.png" />

So because the full name is set on the backend to the key `veteranFullName`, but on the front we access `formData.fullName`, we're not getting any data back to prefill. 

As far as I can tell, if we changed the prefill transformer so it did `veteranFullName: formData.veteranFullName`, that would work 🤞

**Comment by reiting** [comment]
@michaelclement okay. So we would need to also change the keys throughout the form I assume?

**Comment by michaelclement** [comment]
I don't think so - the lefthand keys in the prefill transformer just need to match what the frontend uses/produces, and the right-hand ones need to match what the backend provides. So as long as we update the prefill transformer to pair them up, that should be the only place we need to make changes

**Comment by reiting** [comment]
@marywang2 a PR is awaiting platform approval

**Comment by reiting** [comment]
@marywang2 the PR has been merged.

**Comment by marywang2** [comment]
@reiting I'm not seeing the changes on staging still...let me know if you want to look at it together.

**Comment by reiting** [comment]
@marywang2 At this point, I'm not sure what else could be done on the frontend. There's very little we do for this other than the prefillTransformer. I think it's possible that it's a backend issue instead. @balexandr could you take a look?

**Comment by michaelclement** [comment]
Made some changes and added a some info logging [in this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/29646) so that we can hopefully get some insight into what's happening on staging. As soon as platform approves we can push this to staging and hopefully get some clarity on what's not working.

**Comment by marywang2** [comment]
@reiting @michaelclement Finally was able to test on staging and have a few observations:
- After I log in, it goes directly to the review page (I was expecting it to go to the first page of the form, rather than directly to the review page)
- Home address is missing (it's in my test profile so I would have expected it to be present)
- I'm able to edit the fields within the review page but not the SSN field (cannot edit on the SSN screen either, it won't allow me to save any changes or continue)
- I'm not seeing save in progress indicators
- There should be a pre-fill message as well (does that need to be added to the design or is it already in the forms library)?

**Comment by michaelclement** [comment]
Some notes on what still needs to be done to address Mary's bullets above, I believe this will cover it:
- Update the `returnUrl` value so it points to the first page of the form rather than `review-and-submit`
- Fix prefill to add home address
  - The `prefillTransformer` needs to be updated to add a home address key (could just duplicate the mailing address?) 
  - The backend mapping file (`10-7959F-1.yml`) needs to add the home address for us to use (if it's actually available). I'm seeing evidence of a value `residential_address` that may be available to us and that should be the user's home address (need to talk with Bryan)
  - Research the SSN bug to get to the bottom of it

**Comment by reiting** [comment]
@marywang2 The frontend prefill transformer is updated and ready to go.Currently in team review

**Comment by michaelclement** [comment]
@marywang2 As an update, we're still communicating with platform via the help ticket. In the meantime, Bryan has put together a [new backend PR here](https://github.com/department-of-veterans-affairs/vets-api/pull/16783) that hopefully addresses the home address/mailing address issue.

**Comment by michaelclement** [comment]
@marywang2 Bryan's backend PR got us the correct data, so now we just need [this last adjustment](https://github.com/department-of-veterans-affairs/vets-website/pull/29855) to be merged into the frontend and everything should work.

Waiting on:
- [X] Team approval
- [ ] Platform approval

**Comment by reiting** [comment]
@marywang2 @baltes the last PR for this was merged about 5 hours ago, so it SHOULD be ready to take a look at on staging.

**Comment by michaelclement** [comment]
@marywang2 @reiting 
Looks like the two addresses get returned with different keynames 🤦‍♂️ We assumed that all addresses would use the same properties once returned from backend, but that was not correct:

The reason "City" shows up is because both physical and mailing address use the "City" key. But notice that the first address uses "Street" and the second uses "Address Line 1"
<img width="824" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNHY3QkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b553ff2ff7cd89a74335e95cd2e5bea715302b12/Screenshot%202024-05-21%20at%2009.14.30.png" alt="Screenshot 2024-05-21 at 09.14.30.png" />

**Comment by michaelclement** [comment]
@marywang2 This fix has been merged and should show up on staging in the next couple hours. Thanks!

**Comment by michaelclement** [comment]
Confirmed it's working on staging:

<img width="517" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN0VDQlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c4a1a312a272b545c8062561abb5d7c7563cc4cc/Screenshot%202024-05-22%20at%2012.23.03.png" alt="Screenshot 2024-05-22 at 12.23.03.png" />

**Comment by marywang2** [comment]
@michaelclement looks good, thank you. Should I create another ticket for masking the SSN?

**Comment by michaelclement** [comment]
@marywang2 Yes, let's make that a separate issue - it will require some UI customizing and is more of a presentation issue than a pre-fill thing I'd say
